frontend/src/chain/queries/useValidateAddressQuery.ts

```diff
@@ -0,0 +1,23 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { useAssertWalletCore } from '../../providers/WalletCoreProvider';
+import { AddressServiceFactory } from '../../services/Address/AddressServiceFactory';
+import { ChainAccount } from '../ChainAccount';
+
+export const useValidateAddressQuery = ({ chainId, address }: ChainAccount) => {
+  const walletCore = useAssertWalletCore();
+
+  return useQuery({
+    queryKey: ['validateAddress', chainId, address],
+    queryFn: async () => {
+      const addressService = AddressServiceFactory.createAddressService(
+        chainId,
+        walletCore
+      );
+
+      const isValid = await addressService.validateAddress(address);
+
+      return isValid ? null : 'Invalid address';
```

Comment:
Yeah, we can do that also, but I was thinking to have validation function returning either nothing(`undefined`) or in case of an invalid value - either a error message or throw an Error.